server/src/main/java/org/opensearch/tasks/TaskInfo.java

```diff
@@ -150,6 +185,11 @@ public TaskInfo(StreamInput in) throws IOException {
         } else {
             resourceStats = null;
         }
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            cancelledAt = in.readLong();
+        } else {
+            cancelledAt = -1;
```

Comment:
Isn't it better to use `readOptionalLong` and `writeOptionalLong` and use null to represent the optional-ness of this value as opposed to using a magic `-1` value?

Reply:
Yeah make sense. Will change.